packages/kbn-search-api-panels/types.ts

```diff
@@ -27,22 +27,22 @@ export interface LanguageDefinitionSnippetArguments {
 
 type CodeSnippet = string | ((args: LanguageDefinitionSnippetArguments) => string);
 export interface LanguageDefinition {
+  name: string;
```

Comment:
I’ve made any of the "codeSnippets" as optional. I see there is already a safeguard against undefined values in [utils.ts](https://github.com/elastic/kibana/blob/main/packages/kbn-search-api-panels/utils.ts#L18) so I think this should be OK. I’ve also made `docLink` optional and added a check in `install_client.tsx`. This makes the interface a little more flexible when the `CodeBox` component is used in isolation. This seemed like the approach that required the least refactoring, but open to feedback here.

Reply:
I'm fine with this. I think it would be nice to make the CodeBox a bit more generic in the future, but this is a good middle-step.